tokio-trace/tokio-trace-core/src/subscriber.rs

```diff
@@ -262,6 +264,54 @@ pub trait Subscriber: 'static {
     fn drop_span(&self, id: span::Id) {
         let _ = id;
     }
+
+    // === Downcasting methods ================================================
+
+    /// Gets the `TypeId` of `Self`.
+    ///
+    /// Implementations of `Subscriber` are **not** expected to override this!
+    #[doc(hidden)]
+    fn type_id(&self) -> TypeId {
```

Comment:
Why is this fn needed?

Reply:
Now that I actually think about it, this isn't actually necessary --- all subscribers should implement `Any` (as they're `'static`), so we can use `Any::type_id`. I think this can safely be removed. Thanks for noticing that!